### Paired T-test 

A paired t-test is performed to see if using all features give significantly different predictions than using min, max and ch only

In [1]:
from scipy.stats import ttest_rel
import pandas as pd

In [2]:
df_all = pd.read_csv("predictions_all.csv")
df_minmaxch = pd.read_csv("predictions_minmaxch.csv")

df = df_all.merge(df_minmaxch, on=["record_id", "redcap_repeat_instance", "eye"])

In [4]:
t_stat, p = ttest_rel(df["pred_a"], df["pred_b"])
print(f"t-stat: {t_stat:.4f}, p-value: {p:.4f}")

t-stat: 0.6085, p-value: 0.5429


In [5]:
# Mean absolute difference between the two models
print("Mean abs difference:", round(abs(df["pred_a"] - df["pred_b"]).mean(), 3))

# How often do they give the exact same prediction
print("Exact agreement:", round((df["pred_a"] == df["pred_b"]).mean() * 100, 1), "%")

Mean abs difference: 0.014
Exact agreement: 86.0 %


The paired t-test revealed no statistically significant difference between the two reconstruction models (t = 0.6085, p = 0.5429). The mean absolute difference in predictions was 0.014 NPi units, and the models produced identical predictions in 86% of cases, indicating that they are equivalentt despite the differences in functional form.